api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -280,6 +410,30 @@ default UpdateSchema updateColumn(String name, Type.PrimitiveType newType, Strin
     return updateColumn(name, newType).updateColumnDoc(name, newDoc);
   }
 
+  /**
+   * Update a column in the schema to a new primitive type.
+   *
+   * <p>The name is used to find the column to update using {@link Schema#findField(String)}.
+   *
+   * <p>Only updates that widen types are allowed.
+   *
+   * <p>Columns may be updated and renamed in the same schema update.
+   *
+   * @param name name of the column to rename
+   * @param newType replacement type for the column
+   * @param newDoc replacement documentation string for the column
+   * @return this for method chaining
+   * @throws IllegalArgumentException If name doesn't identify a column in the schema or if this
+   *     change introduces a type incompatibility or if it conflicts with other additions, renames,
+   *     or updates.
+   */
+  default UpdateSchema updateColumn(
```

Comment:
I'm not sure I follow why we would need type widening to also include setting a new default value.  Couldn't these be done separately?

Reply:
The previous version of this method is only used in tests, so I think you're right that we don't really need to have this.